Sources/StreamVideoSwiftUI/Utils/PictureInPicture/StreamPictureInPIctureView.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  StreamPictureInPIctureView.swift
+//  StreamVideoSwiftUI
+//
+//  Created by Ilias Pavlidakis on 9/1/24.
+//
+
+import Foundation
+import SwiftUI
+
+struct StreamPictureInPictureView: UIViewRepresentable {
+
+    func makeUIView(context: Context) -> UIView {
+        let view = UIView()
+        view.backgroundColor = .clear
+        if #available(iOS 15.0, *) {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+                StreamPictureInPictureAdapter.shared.sourceView = view
+            }
+        }
+        return view
+    }
+
+    func updateUIView(_ uiView: UIView, context: Context) {
+        if #available(iOS 15.0, *) {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 1) {
+                StreamPictureInPictureAdapter.shared.sourceView = uiView
+            }
+        }
+    }
+}
+
+struct PictureInPictureModifier: ViewModifier {
+
+    func body(content: Content) -> some View {
+        content
+            .background(StreamPictureInPictureView())
+    }
+}
+
+extension View {
+
+    @ViewBuilder
+    public func enablePictureInPicture() -> some View {
+        self.modifier(PictureInPictureModifier())
+    }
+}
+
+import UIKit
+
+extension UIApplication {
```

Comment:
Why do we need this gorgeous code? 😅

Reply:
We didn't at the end :P I hated it and i'm very happy i got to remove it haha